### DISCUSSION TITLE
Joint text2mel / vocoder training

### DISCUSSION THREAD

**Discussion by lheuveline** [coqui-ai/TTS#1157] [General Q&A] [answered]
Hi !

I'm working with Espnet from few months now, and I'm really interested by moving to CoquiTTS.
I'm investigating FS2 - Hifigan joint training. Is there any way in CoquiTTS to achieve joint training to avoid feature mismatch ?

**Comment by erogol** [comment, marked as answer]
Hey welcome to 🐸

You can train them one by one to extract features from FS2 by using ```TTS/bin/extract_tts_spectrograms.py``` and passing them to the HifiGAN. 

However, we don't have a unified interface yet.  If you will, you can implement it on your own regarding the VITS implementation.
